ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/identity_provider_config_service.proto

```diff
@@ -0,0 +1,131 @@
+//
+// An identity provider config resource, once it has been created, can be modified.
+// In order to update the properties represented by the ``IdentityProviderConfig`` message use the ``UpdateIdentityProviderConfig`` RPC.
+// The only fields that can be modified are those marked as ``Modifiable``.
+service IdentityProviderConfigService {
+
+  // Create a new identity provider configuration.
+  rpc CreateIdentityProviderConfig (CreateIdentityProviderConfigRequest) returns (CreateIdentityProviderConfigResponse);
+
+  // Get the identity provider configuration data by id.
+  rpc GetIdentityProviderConfig (GetIdentityProviderConfigRequest) returns (GetIdentityProviderConfigResponse);
+
+  // Update selected modifiable attribute of a identity provider config resource described
+  // by the ``IdentityProviderConfig`` message.
+  rpc UpdateIdentityProviderConfig (UpdateIdentityProviderConfigRequest) returns (UpdateIdentityProviderConfigResponse);
+
+  // List all existing identity provider configurations.
+  rpc ListIdentityProviderConfigs (ListIdentityProviderConfigsRequest) returns (ListIdentityProviderConfigsResponse);
+
+  // Delete an existing identity provider configuration.
+  rpc DeleteIdentityProviderConfig (DeleteIdentityProviderConfigRequest) returns (DeleteIdentityProviderConfigResponse);
+
+}
+
+message IdentityProviderConfig {
+  // The identity provider identifier
+  // Must be a valid LedgerString (as describe in ``value.proto``).
+  // Required
+  string identity_provider_id = 1;
+
+  // When set, then the user identified by the identity provider is denied all access to the Ledger API.
+  // Otherwise, the user identified by the identity provider has access to the Ledger API as per the user's record.
+  // Optional,
+  // Modifiable
+  bool is_deactivated = 2;
+
+  // Specifies the issuer of the JWT token that should be used in the "iss" claim of a JWT token to instruct the
+  // participant node to use this identity provider to authenticate the token.
+  // The issuer value is a case sensitive URL using the https scheme that contains scheme, host,
+  // and optionally, port number and path components and no query or fragment components.
+  // Required
+  // Modifiable
+  string issuer = 3;
+
+  // The JWKS (JSON Web Key Set) URL.
+  // The Ledger API will expect all tokens to be signed with RS256 (RSA Signature with SHA-256) with the public
+  // key loaded from the given URL.
+  // Required
+  // Modifiable
+  string jwks_url = 4;
+}
+
+
+message CreateIdentityProviderConfigRequest {
+  // Required
+  IdentityProviderConfig identity_provider_config = 1;
+}
+
+message CreateIdentityProviderConfigResponse {
+  IdentityProviderConfig identity_provider_config = 1;
+}
+
+message GetIdentityProviderConfigRequest {
+  // Required
+  string identity_provider_id = 1;
+}
+
+message GetIdentityProviderConfigResponse {
+  IdentityProviderConfig identity_provider_config = 1;
+}
+
+message ListIdentityProviderConfigsRequest {
+
```

Comment:
Add support for pagination or mark it as tech debt.

Reply:
@mziolekda actually asked to remove the pagination as we won't be supporting for more than 16 IDPs, so pagination would be an overkill here

Reply:
I concur, supporting additional IDPs puts a stress on the authorization mechanism that we do not want. Limiting the number of IDPs is better.

Reply:
Do you think we need to note it as well in the doc of the proto?

Reply:
I believe it would be good to add a note, as this way future devs know that they'll need to  add pagination once that limit of 16 IDPs is removed.